source/sphinx_extensions/command_block/extension.py

```diff
@@ -40,17 +63,34 @@ class CommandBlockDirective(docutils.parsers.rst.Directive):
     has_content = True
 
     option_spec = {
-        'no-exec': docutils.parsers.rst.directives.flag
+        'no-exec': docutils.parsers.rst.directives.flag,
+        'url': docutils.parsers.rst.directives.unchanged_required,
+        'rename': docutils.parsers.rst.directives.unchanged_required,
     }
 
     def run(self):
-        self.assert_has_content()
-        commands = functools.reduce(self._parse_multiline_commands,
-                                    self.content, [])
-
-        nodes = [
-            self._get_literal_block_node(self.content)
-        ]
+        command_mode = True if self.name == 'command-block' else False
+        opts = self.options
+        download_opts = [k in opts for k in ['url', 'rename']]
+
+        if command_mode:
+            self.assert_has_content()
+            if any(download_opts):
+                raise self.error('command-block does not support the '
```

Comment:
![image](http://25.media.tumblr.com/643560e9366158f23fdd73e21db57cff/tumblr_mhrpirR3OQ1r6g9rho1_400.gif)

Oh I meant to chat with you about this! I don't think that `sphinx.errors.ExtensionError` is the right error to use here. The `CommandBlockDirective` is not sphinx-based, but rather, docutils-based. Cruising through the docutils [docs](http://docutils.sourceforge.net/docs/howto/rst-directives.html#error-handling), this looks like the right way to raise exceptions related to custom nodes/directives. Also, the trackback is a bit more informative:

```sh
(q2dev) ☁  docs [issue-65] ⚡ make preview
rm -rf build/*
sphinx-build -b dirhtml -D command_block_no_exec=1 -d build/doctrees -W source build/preview
Running Sphinx v1.4.6
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [dirhtml]: targets for 27 source files that are out of date
updating environment: 85 added, 0 changed, 0 removed
reading sources... [100%] tutorials/moving-pictures

Warning, treated as error:
/Users/matthew/caporaso/qiime2/docs/source/interfaces/q2studio.rst:14: ERROR: Missing options for the download directive. Please specify `url` and `rename`.

make: *** [preview] Error 1
```

(note the file and line number leading you to the source of the error)

vs

```sh
(q2dev) ☁  docs [issue-65] ⚡ make preview
rm -rf build/*
sphinx-build -b dirhtml -D command_block_no_exec=1 -d build/doctrees -W source build/preview
Running Sphinx v1.4.6
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [dirhtml]: targets for 27 source files that are out of date
updating environment: 85 added, 0 changed, 0 removed
reading sources... [ 16%] interfaces/q2studio
Extension error:
Missing options for the download directive. Please specify `url` and `rename`.
make: *** [preview] Error 1
```

The [docs](http://www.sphinx-doc.org/en/1.5.1/extdev/appapi.html#sphinx.errors.ExtensionError) for the `sphinx.errors.ExtensionError` indicate that it should be "Used for errors in setting up extensions," which from my perspective isn't really what is going on in the `run` method --- personally I think we should update the existing uses of `ExtensionError` to use the docutils-based exception, instead. If you agree, I can do that in this PR, or save it for a 🌧 day.